### PR TITLE
Add cypress to institution service

### PR DESCRIPTION
**Pull request by tphalp**
- Add cypress module, along with needed folder structure, config, npm run commands, etc.
- Adds e2e test for 'ping' endpoint
- Remove need for dotenv, and hard-code PORT

### CODE REVIEW

File: apps/institution-service/cypress/support/commands.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/// <reference types="cypress" />
+import "@testing-library/cypress/add-commands";
```
Comment (wesrisenmay-mx) [inline apps/institution-service/cypress/support/commands.ts:2]: We need the following:

1. github action that runs this test suite
2. At minimum a fake cypress test, or add a ping endpoint and ping that
3. package.json commands to open and run cypress

Comment (tphalp) [reply apps/institution-service/cypress/support/commands.ts:2]: I'll add 1 and 2. For 3, these commands already existed in the package file, so weren't necessary to add in this PR.

File: apps/institution-service/cypress.config.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { defineConfig } from "cypress";
+import "dotenv/config";
+
+export default defineConfig({
+  env: {
+    PORT: process.env.PORT || 8088,
```
Comment (wesrisenmay-mx) [inline apps/institution-service/cypress.config.ts:6]: Why don't we just hardcode a port?

Comment (tphalp) [reply apps/institution-service/cypress.config.ts:6]: Because I took this from the auth service, and that's how it was done there.

Comment (tphalp) [reply apps/institution-service/cypress.config.ts:6]: Oh, actually, that's not true. I'll hard-code it.

File: .github/workflows/institution-e2e-tests.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      - run: npm ci
+
+      - name: 'Create env file'
+        run: |
+          ENV_FILE_PATH=./apps/institution-service/.env
+          touch ${ENV_FILE_PATH}
+          
+          # Vars
+          echo -e "${{ needs.setup-env.outputs.env_vars }}" >> ${ENV_FILE_PATH}
+          echo PORT="8088" >> ${ENV_FILE_PATH}
```
Comment (wesrisenmay-mx) [inline .github/workflows/institution-e2e-tests.yml:33]: Need to remove this. Probably in the auth service action too

Comment (tphalp) [reply .github/workflows/institution-e2e-tests.yml:33]: Let's talk about how we decide on port numbers. The auth service uses .env for this. We can do static, but we should decide the best way to proceed. Do they need to be different, or does that not matter, etc.

File: .github/workflows/institution-e2e-tests.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+name: Institution E2E
+
+on: pull_request
+
+jobs:
+  setup-env:
```
Comment (wesrisenmay-mx) [inline .github/workflows/institution-e2e-tests.yml:6]: We don't need this anymore right?

File: apps/institution-service/app.ts [hunk 5]
```diff
@@ -1,11 +1,9 @@
-import dotenv from "dotenv";
 import express, { Request, Response } from "express";
 import logger from "morgan";
 import institutionRoutes from "./routes/institutionRoutes";
 
-dotenv.config();
 const app = express();
-const PORT = process.env.PORT || 3030;
+const PORT = process.env.PORT || 8088;
```
Comment (wesrisenmay-mx) [inline apps/institution-service/app.ts:6]: I'd rather it fails if it's misconfigured

**Review by wesrisenmay-mx — APPROVED** [review]

**Pull request merged**